bindings/python/tests/test_reader_datetime.py

```diff
@@ -74,13 +66,149 @@ def test_read_with_timestamps(self):
         device = instance.add_device('daqref://device0')
         signal = device.signals_recursive[0]
         stream = opendaq.StreamReader(signal)
-        reader = opendaq.TimeReader(stream)
+        reader = opendaq.TimeStreamReader(stream)
 
         time.sleep(0.1)
         (values, domain) = reader.read_with_timestamps(100)
         for t in domain:
             self.assertIsInstance(t, numpy.datetime64)
 
+    def test_tail_read(self):
+        import time
+
+        instance = opendaq.Instance()
```

Comment:
Calling opendaq.Instance can be expensive, because it loads modules. I would suggest the following approach:

Expose "CreateMockSignal" function to python from the bindings. This returns a signal wrapper which has the following functions:

- getSignal,
- getDomainSignal,
- addData

When calling init of a wrapper this:

- creates a value signal and a domain signal
- sets domainSignal property of value signal to domainSignal
- configures both signals with signal descriptor (data type can be parameter to init)

AddData method:

- takes py:array parameter
- constructs value and domain packet
- copies data into packet
- calls ISignal.addPacket on both value and domain packets.

You can then pass mockSignal.getSignal() to reader factories. Mock signal should be used in tests for readers, function blocks etc...